lib/dfe_sign_in/api.rb

```diff
@@ -0,0 +1,43 @@
+require "dfe_sign_in/api/request"
+require "dfe_sign_in/api/response"
+
+module DfeSignIn
+  module API
```

Comment:
The original API code has been completely restructured into different modules/classes.